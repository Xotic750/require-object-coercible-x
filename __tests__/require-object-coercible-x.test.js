import noop from 'lodash/noop';
import requireObjectCoercible from 'src/require-object-coercible-x';

/* eslint-disable-next-line compat/compat */
const hasSymbol = typeof Symbol === 'function' && typeof Symbol('') === 'symbol';
const ifSymbolIt = hasSymbol ? it : xit;

describe('basic tests', function() {
  it('should throw TypeError everything', function() {
    expect.assertions(3);
    expect(function() {
      requireObjectCoercible();
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      requireObjectCoercible(undefined);
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      requireObjectCoercible(null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should return value for everything', function() {
    expect.assertions(1);
    const values = [true, 'abc', 1, noop, [], /r/];
    const actual = values.map(requireObjectCoercible);
    expect(actual).toStrictEqual(values);
  });

  ifSymbolIt('should return Symbol', function() {
    expect.assertions(1);
    /* eslint-disable-next-line compat/compat */
    const sym = Symbol('foo');
    expect(requireObjectCoercible(sym)).toBe(sym);
  });
});
