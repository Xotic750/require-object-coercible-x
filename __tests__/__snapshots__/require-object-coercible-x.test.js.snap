// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic tests should throw TypeError everything 1`] = `"Cannot call method on undefined"`;

exports[`basic tests should throw TypeError everything 2`] = `"Cannot call method on undefined"`;

exports[`basic tests should throw TypeError everything 3`] = `"Cannot call method on null"`;
